XForm/XForm/Functions/Coalesce.cs

```diff
@@ -0,0 +1,172 @@
+﻿using System;
+using System.Collections.Generic;
+using XForm.Data;
+using XForm.Query;
+
+namespace XForm.Functions
+{
+    public class Coalesce : IFunctionBuilder
+    {
+        public Type ReturnType => null;
+
+        public string Usage => "Coalesce({Col|Func|Const}, ...)";
+
+        public string Name => "Coalesce";
+
+        public IXColumn Build(IXTable source, XDatabaseContext context)
+        {
+            return new CoalesceTransformFunction(source, context);
+        }
+    }
+
+    public class CoalesceTransformFunction : IXColumn
+    {
+        private List<IXColumn> _columns = new List<IXColumn>();
+        private ICoalescer _coalescer;
+
+        public CoalesceTransformFunction(IXTable source, XDatabaseContext context)
+        {
+            while (context.Parser.HasAnotherArgument)
+            {
+                IXColumn column = context.Parser.NextColumn(source, context);
+                _columns.Add(column);
+                if (System.String.IsNullOrEmpty(column.ColumnDetails.Name)) throw new ArgumentException($"Column {_columns.Count} passed to 'Column' wasn't assigned a name. Use 'AS [Name]' to assign names to every column selected.");
+            }
+
+            if (!AllTypesMatch(_columns))
+                throw new UsageException("Coalesce requires all column types to match");
+
+            _coalescer = (ICoalescer)Allocator.ConstructGenericOf(typeof(Coalescer<>), _columns[0].ColumnDetails.Type);
+
+            ColumnDetails = new ColumnDetails($"{_columns[0].ColumnDetails.Name}.Coalesce", _columns[0].ColumnDetails.Type);
```

Comment:
I'm not sure what the default column name should be, but matching any of the inputs individually feels a little weird. Maybe it should just default to "Coalesce", and then if you have more than one Coalesce call in the select list you'll have to assign them each column names with an outer "AS"